File: app/src/main/java/com/smartpack/kernelprofiler/utils/OnBootReceiver.java
package com.smartpack.kernelprofiler.utils;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

/*
 * Created by sunilpaulmathew <sunil.kde@gmail.com> on May 22, 2020
 */

public class OnBootReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        if (Intent.ACTION_BOOT_COMPLETED.equals(intent.getAction())) {
            if (Utils.rootAccess()) {
                Utils.startService(context, new Intent(context, BootService.class));
            }
        }
    }
}

File: app/src/main/java/com/smartpack/kernelprofiler/utils/OnBootReceiver.kt
package com.smartpack.kernelprofiler.utils

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.smartpack.kernelprofiler.utils.root.RootUtils

/**
 * Created by sunilpaulmathew <sunil.kde></sunil.kde>@gmail.com> on May 22, 2020
 */
class OnBootReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        if (Intent.ACTION_BOOT_COMPLETED == intent.action) {
            if (RootUtils.rootAccess()) {
                Utils.startService(context, Intent(context, BootService::class.java))
            }
        }
    }
}